datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java

```diff
@@ -313,12 +313,23 @@ public void run() {
       while (!isInterrupted() && !_shutdown) {
         try {
           List<String> newPartitionInfo = getPartitionsInfo(consumer);
-          LOG.debug("Fetch partition info for {}, oldPartitionInfo: {}, new Partition info: {}"
-              , datastream.getName(), _subscribedPartitions, newPartitionInfo);
+          LOG.debug("Fetch Partitions Info for {}, old Partitions Info: {}, new Partitions Info: {}",
+              datastream.getName(), _subscribedPartitions, newPartitionInfo);
 
           if (!ListUtils.isEqualList(newPartitionInfo, _subscribedPartitions)) {
-            LOG.info("Get updated partition info for {}, oldPartitionInfo: {}, new Partition info: {}"
-                , datastream.getName(), _subscribedPartitions, newPartitionInfo);
+            LOG.info("Get updated Partitions Info for {}, old Partitions Info: {}, new Partitions Info: {}",
+                datastream.getName(), _subscribedPartitions, newPartitionInfo);
+
+            List<String> addedTopicPartitions = newPartitionInfo
+                .stream()
+                .filter(topicPartition -> !_subscribedPartitions.contains(topicPartition))
```

Comment:
Converting these list into set, cache the set as well and doing:

added = NewSet - ExistingSet or NewSet.removeall(existingSet)
removed = ExistingSet - NewSet or ExistingSet.removeall(NewSet)

will make this operation faster. Right now its O(N^2) complexity.

Reply:
Modified to use a set, found the intersection of the two sets and removed the intersection from each.